Utils/Dataflow/030_PDFAnalyzer/xmltable.py

```diff
@@ -260,8 +257,8 @@ def row_text(self, num=None, row=False):
         if num is not None or num == 0:
             row = self.rows[num]
         text = ""
-        for l in row:
-            text += l.text + "!"
+        for line in row:
```

Comment:
For this we have map + join: return ("!".join(map(lambda r: r.text), row) + "!")

Not sure, by the way, that the trailing '!' is needed: it contradicts with the function description, since it says "between lines".

Reply:
It seems that row should be the second argument of map in your example. I think i'll use [A for B in C] instead of map, since that construct is more familiar to me.

Comment about trailing "!" makes sense, thanks.

Reply:
return ("!".join(map(lambda r: r.text, row)) + "!"), exactly.